### PR TITLE
Update StatusCake provider

### PR DESCRIPTION
**Pull request by smneal**
### Context
Update the StatusCake provider to 2.0.3 to use the latest version of the API. This contains breaking changes.

### Changes proposed in this pull request
1. Update the StatusCake alert resource to be compatible with the new API
2. Update the Terraform StatusCake provider to 2.0.3
3. Add a basic auth section for staging environment alerts
4. Add variable to support toggle on/off basic auth tests

### Guidance to review
Confirmed the new resource works by deploying to the review app.

### CODE REVIEW

**Comment by smneal** [comment]
[Update StatusCake terraform provider](https://trello.com/c/96qh0O74/658-update-statuscake-terraform-provider)

File: terraform/paas/staging.env.tfvars [hunk 1]
```diff
@@ -12,15 +12,3 @@ application_environment   = "dfe-school-experience-staging"
...
-    website_name   = "School Experience (Staging)"
-    website_url    = "https://school-experience-app-staging.london.cloudapps.digital/healthcheck.json"
-    test_type      = "HTTP"
-    check_rate     = 60
-    contact_group  = [239498]
-    trigger_rate   = 0
-    confirmations  = 2
-    node_locations = ["UKINT", "UK1", "MAN1", "MAN5", "DUB2"]
-  }
-}
```
Comment (RMcVelia) [inline terraform/paas/staging.env.tfvars:26]: Do we want to remove staging statuscake?

Comment (smneal) [reply terraform/paas/staging.env.tfvars:26]: Adding back in with basic auth

**Review by RMcVelia — APPROVED** [review]

File: terraform/paas/statuscake.tf [hunk 2]
```diff
@@ -1,13 +1,68 @@
...
+      "521",
+      "522",
+      "523",
+      "524",
+      "520",
+      "598",
+      "599"
+    ]
+    dynamic "basic_authentication" {
+      for_each = var.alerts_basic_authentication == true ? [1] : []
```
Comment (saliceti) [inline terraform/paas/statuscake.tf:57]: For readability, can we try a boolean-styled var name such as:
```
for_each = var.statuscake_enable_basic_auth ? [1] : []
```

Comment (smneal) [reply terraform/paas/statuscake.tf]: Addressed

**Pull request merged**